File: 1. Learn the Basics/4. Hashing/Python/2. Find highest and lowest frequency element.py
# cook your dish here
#Find highest and lowest frequency of array elements using hashing


def frequency(arr,n):
    mp = {}
    for num in arr:
        if num in mp:
            mp[num] += 1
        else:
            mp[num] = 1
    return mp

n = int(input())
arr = list(map(int,input().strip().split()))
mp = frequency(arr,n)
max_key = 0
min_key = 0
max_value = -999999
min_value = 999999
for key in mp:
    print(key,mp[key])
    if(mp[key]>max_value):
        max_value = mp[key]
        max_key = key
    if(mp[key]<min_value):
        min_value = mp[key]
        min_key = key

print("Mximum Value ",max_key,max_value)
print("Minimum Value ",min_key,min_value)

q = int(input())
while(q>0):
    number = int(input())
    if number in mp:
        print(number,mp[number])
    else:
        print(number,0)
    q = q-1

File: 1. Learn the Basics/4. Hashing/Java/2. Find highest and lowest frequency element.java
/* package codechef; // don't place package name! */

import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */

/*
Find highest and lowest frequency of array elements using hashing
*/

class Codechef
{
    
	public static void main (String[] args) throws java.lang.Exception
	{
		// your code goes here
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int arr[] = new int[n];
		Map<Integer,Integer> mp = new HashMap<>();
		for(int i=0;i<n;i++){
		    arr[i] = sc.nextInt();
		    if(mp.containsKey(arr[i])){
		        mp.put(arr[i],mp.get(arr[i])+1);
		    } else {
		        mp.put(arr[i],1);
		    }
		}
	    int max_key=0,min_key=0,max_value=Integer.MIN_VALUE,min_value=Integer.MAX_VALUE;
	    
		for(Map.Entry<Integer,Integer> entry : mp.entrySet()){
		    System.out.println(entry.getKey()+"->"+entry.getValue());
		    if(entry.getValue()>max_value){
		        max_value = entry.getValue();
		        max_key = entry.getKey();
		    }
		    if(entry.getValue() < min_value){
		        min_value = entry.getValue();
		        min_key = entry.getKey();
		    }
		}
		System.out.println("Maximum "+max_key+" "+max_value);
		System.out.println("Minimum "+min_key+" "+min_value);
		
		int q = sc.nextInt();
		while(q>0){
		    int number = sc.nextInt();
		    if(mp.containsKey(number)){
		        System.out.println(mp.get(number));
		    }else{
		        System.out.println(0);
		    }
		    q--;
		}
	}
}
